org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/utils/XMLGenerator.java

```diff
@@ -472,8 +510,15 @@ public static MarkupContent createMarkupContent(CMElementDeclaration cmElement,
 	 * @return true if the element child element is to be generated.
 	 */
 	private boolean isGenerateChild(CMElementDeclaration elementDeclaration, boolean generateOnlyRequired,
-			String childName) {
-		if (!generateOnlyRequired || (!elementDeclaration.isOptional(childName) && generateOnlyRequired)) {
+			CMElementDeclaration child) {
+		// With RelaxNG, no need to generate if the child element is the second element
+		// in <choice>
+		if (elementDeclaration instanceof CMRelaxNGElementDeclaration

```

Comment:
Please don't write specific usecase in a generic component. With the idea of getRequiredElements, you should not need this isGenerateChild method.

I think your code covers the bug with choice when all elements are generated. It is because getElements() need to be improved, so let's keep this bug.